File: src/PHP/BehavioralPatterns/Mediator/ChainReaction/Participant.php
<?php

namespace BehavioralPatterns\Mediator\ChainReaction;

abstract class Participant {
    protected Mediator $mediator;

    public function __construct(Mediator $mediator) {
        $this->mediator = $mediator;
    }

    protected function getMediator(): Mediator
    {
        return $this->mediator;
    }
}

File: src/C#/BehavioralPatterns/Mediator/ChainReaction/Participant.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ChainReaction
{
    public abstract class Participant
    {
        protected Participant(Mediator mediator)
        {
            Mediator = mediator;
        }

        protected Mediator Mediator { get; private set; }
    }
}
